apps/dev-api/domains/user/utils/testSchema/index.js

```diff
@@ -0,0 +1,214 @@
+        dv: 1,
+        sender,
+        phone: createTestPhone(),
+        code: faker.random.numeric(CONFIRM_ACTION_CODE_LENGTH),
+        expiresAt: dayjs().add(CONFIRM_ACTION_TTL_IN_SEC, 's'),
+        ...extraAttrs,
+    }
+    const obj = await ConfirmPhoneAction.create(client, attrs)
+    return [obj, attrs]
+}
+
+async function updateTestConfirmPhoneAction (client, id, extraAttrs = {}) {
+    if (!client) throw new Error('no client')
+    if (!id) throw new Error('no id')
+    const sender = { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+
+    const attrs = {
+        dv: 1,
+        sender,
+        ...extraAttrs,
+    }
+    const obj = await ConfirmPhoneAction.update(client, id, attrs)
+    return [obj, attrs]
+}
+
+async function authenticateUserWithPhoneAndPasswordByTestClient(client, attrs = {}) {
+    if (!client) throw new Error('no client')
+
+    const { data, errors } = await client.mutate(AUTHENTICATE_USER_WITH_PHONE_AND_PASSWORD_MUTATION, { data: attrs })
+    throwIfError(data, errors)
+    return [data.result, attrs]
+}
+/* AUTOGENERATE MARKER <FACTORY> */
+
+async function startConfirmPhoneActionByTestClient(attrs = {}, client) {
+    client ??= await makeClient()
+    attrs.dv ??= 1
+    attrs.sender ??= { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+    attrs.phone ??= createTestPhone()
+
+    const { data, errors } = await client.mutate(START_CONFIRM_PHONE_ACTION_MUTATION, {
+        data: attrs
+    })
+    throwIfError(data, errors)
+
+    return [data.result, attrs]
+}
+
+async function completeConfirmPhoneActionByTestClient(id, attrs = {}, client) {
+    client ??= await makeClient()
+    attrs.dv ??= 1
+    attrs.sender ??= { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+    attrs.actionId = id
+    attrs.code ??= faker.random.numeric(CONFIRM_ACTION_CODE_LENGTH)
+
+    const { data, errors } = await client.mutate(COMPLETE_CONFIRM_PHONE_ACTION_MUTATION, {
+        data: attrs,
+    })
+    throwIfError(data, errors)
+
+    return [data.result, attrs]
+}
+
+async function registerNewTestUser (userAttrs = {}, client) {
+    // NOTE: needed to read confirmation code
+    const adminClient = await makeLoggedInAdminClient()
+
+    userAttrs.sender ??= { dv: 1, fingerprint: faker.random.alphaNumeric(8) }
+    userAttrs.password ??= faker.internet.password()
+    userAttrs.phone ??= createTestPhone()
+    userAttrs.name ??= faker.internet.userName()
+    client ??= await makeClient()
+
+    let { sender, name, phone, password, actionId } = userAttrs
+
+    if (!actionId) {
+         [{actionId}] = await startConfirmPhoneActionByTestClient({
+            sender,
+            phone,
+        }, client)
+
+        const { code } = await ConfirmPhoneAction.getOne(adminClient, {id: actionId})
+
+        await completeConfirmPhoneActionByTestClient(actionId, {
+            sender,
+            code,
+        }, client)
+    }
+
+
+    const { data, errors } = await client.mutate(REGISTER_NEW_USER_MUTATION, {
```

Comment:
There're few reasons for that:
1. I didn't want to make `ConfirmPhoneAction` visible for anyone initially. Some tests requires me to view and update it as admin, so I've added `read` and `update specific fields` option to it. Creating it by bypassing `startConfirmPhoneAction` mutation should be prohibited.
2. I want tests to be as close to real-life scenarios as possible, that's why I've decided to recreate registration in-depth (instead of recreating complex step from admin user)